### PR TITLE
[MNT] add coverage for utils._aa_str_to_letter, _pseaac_utils, _aptatrans_utils

### LINKED ISSUE #725: [TST] Small utils coverage wins
Part of #719 (test-coverage meta).

Small, self-contained coverage gaps in `pyaptamer/utils/` — good first issues,
each an easy win with `--cov-report=term-missing` pointing straight at the lines.

## Scope
- [ ] `_aa_str_to_letter.py` (67%) — missing line 43
- [ ] `_pseaac_utils.py` (81%) — missing lines 44-45, 48
- [ ] `_aptatrans_utils.py` (86%) — missing lines 88-91, 98, 119

Each is a small pure-function path; add targeted unit tests in
`pyaptamer/utils/tests/`.

### PR DESCRIPTION
**Pull request by siddhanttiwari19**
#### Reference Issues/PRs
Fixes #725

#### What does this implement/fix? Explain your changes.
Adds targeted unit tests closing the coverage gaps described in #725:
- `_aa_str_to_letter.py`: known code, lowercase input, unknown fallback
- `_pseaac_utils.py`: valid sequence (no warning), invalid chars (replaced + warns)
- `_aptatrans_utils.py`: docstring example, empty input, mid-loop `seq_max_len` split, and unmatched-character skip path

Note: at time of writing, `_pseaac_utils.py` and `_aptatrans_utils.py` had no existing tests and showed 0% coverage on `main`, rather than the 81%/86% described in the original issue — these tests now bring both to 100%. `_aa_str_to_letter.py` matched the issue exactly (67%, missing line 43) and is now also 100%.

Verified with: `pytest pyaptamer/utils --cov=pyaptamer/utils --cov-report=term-missing`
All three target files now at 100% coverage, 40/40 tests passing.

#### What should a reviewer concentrate their feedback on?
- Whether the added test cases adequately cover edge cases (empty input, unmatched characters, mid-loop split) or if additional cases would strengthen the suite
- Naming/structure conventions for the new test functions, to confirm they match the rest of the `pyaptamer/utils` test suite

#### Did you add any tests for the change?
Yes — this PR is exclusively test additions (no production code changes). See the file-by-file breakdown above.

#### Any other comments?
Currently addressing a failing `code-quality` check flagged in review — will push a follow-up commit once resolved locally with pre-commit.

#### PR checklist
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] — **note:** current title `test: add coverage for...` doesn't match this convention; should likely be renamed to start with `[MNT]` (test/framework additions) — see note below
- [x] Added/modified tests
- [x] Used pre-commit hooks when committing to ensure that code is compliant with hooks — **this is very likely the cause of your failing code-quality check**; run `pre-commit install` then `pre-commit run --all-files` before your next push

### CODE REVIEW

**Comment by siddharth7113** [comment]
Please follow the PR template while creating PRs, code quality checks are failing.

File: pyaptamer/utils/tests/test_aptatrans_utils.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+﻿"""Tests for seq2vec."""
+
+import numpy as np
+
+from pyaptamer.utils._aptatrans_utils import seq2vec
+
+
+def test_seq2vec_matches_docstring_example():
```
Comment (siddharth7113) [inline pyaptamer/utils/tests/test_aptatrans_utils.py]: this is already a doc-test we don't need this.

**Review by siddharth7113 — CHANGES_REQUESTED** [review]

File: pyaptamer/utils/tests/test_aptatrans_utils.py (hunk 1 above)
Comment (siddhanttiwari19) [reply pyaptamer/utils/tests/test_aptatrans_utils.py]: removed

**Comment by siddhanttiwari19** [comment]
Hi @siddharth7113 ,
I've addressed the requested changes by removing the redundant test and have updated the PR accordingly. I also ran the formatting fixes. Could you please take another look when you have a chance? If everything looks good now, I'd appreciate it if you could approve and merge the PR once the required checks complete.
Thank you!

**Comment by siddhanttiwari19** [comment]
Hi @siddharth7113 , just checking in ,I've addressed the requested change (removed the redundant test) and pushed the formatting fixes. Could you re review when you get a chance? Also looks like the CI workflows are pending your approval to run. Thanks!

**Review by siddharth7113 — CHANGES_REQUESTED** [review]
Can you add docstrings to your tests?

File: pyaptamer/utils/tests/test_aptatrans_utils.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+
+def test_seq2vec_splits_sequences_longer_than_seq_max_len():
+    words = {"A": 1}
+    sequences = (["AAAAA"], ["HHHHH"])
+
+    seq_out, ss_out = seq2vec(sequences, words, seq_max_len=2)
+
+    assert seq_out.shape[1] == 2
+    assert seq_out.shape[0] == 3
```
Comment (siddharth7113) [inline pyaptamer/utils/tests/test_aptatrans_utils.py]: Can you try instead this sequence?  
`seq_out, ss_out = seq2vec((["ACACA"], ["HBEGI"]), {"A": 1, "C": 2}, seq_max_len=2)`

and instead of shape check, assert the exact output
```
    np.testing.assert_array_equal(seq_out, [[1, 2], [1, 2], [1, 0]])
    np.testing.assert_array_equal(ss_out,  [[1, 2], [3, 4], [5, 0]])
```

File: pyaptamer/utils/tests/test_aptatrans_utils.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+
+def test_seq2vec_skips_unmatched_characters():
+    # "Z" is not in the vocabulary at word lengths 1-3, forcing the
+    # "skip character if no match found" branch (i += 1) to execute.
+    words = {"A": 1}
```
Comment (siddharth7113) [inline pyaptamer/utils/tests/test_aptatrans_utils.py]: same as above.

File: pyaptamer/utils/tests/test_aptatrans_utils.py (hunk 2 above)
Comment (siddhanttiwari19) [reply pyaptamer/utils/tests/test_aptatrans_utils.py]: ohk, updated to use your exact sequence and assertions. Pushed now.

File: pyaptamer/utils/tests/test_aptatrans_utils.py (hunk 3 above)
Comment (siddhanttiwari19) [reply pyaptamer/utils/tests/test_aptatrans_utils.py]: Applied the same approach here . Pushed now.

**Comment by siddhanttiwari19** [comment]
All changes has been done let me know if u want to correct anything else @siddharth7113

**Comment by siddhanttiwari19** [comment]
Hi @siddharth7113 , I’ve addressed all the requested changes and updated the PR. Could you please take another look when you have time? If everything looks good, I'd appreciate it if you could approve and merge it. Thank you!

**Comment by siddharth7113** [comment]
Hi @siddhanttiwari19 , Apologies this took so long, but if the test passes this would be good to merge. Also try to use the re-request for review, sometimes I miss the message.
image

**Review by siddharth7113 — APPROVED** [review]
